game.py

```diff
@@ -477,6 +472,10 @@ def _information_per_play(self, action: Action, acting_player_id: int) -> float:
                 total_info += 1
         
         return total_info / 2
+    
+    def retrieve_metric_dict(self) -> dict[str, list]:
+        "Function that allows retrival of metrics dictionary to display results once the game has finished"
+        return self._metric_dict
```

Comment:
Python's [property](https://docs.python.org/3/library/functions.html#property.getter) would make this cleaner.
```suggestion
    @property
    def metric_dict(self) -> dict[str, list]:
        "Retrieve a dict of metrics to display results once the game has finished"
        return self._metric_dict
```